### PR TITLE
Add path web app run

### PR DESCRIPTION
**Pull request by SamirPS**
## What do these changes do?

I modify the web.py file and add Union[str,pathlib.Path] for path

## Are there changes in behavior for the user?

For the option of path, we can have str or pathlib.path

## Related issue number

#6839 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by Dreamsorcerer** [comment]
Need to see a test covering this. Could probably just remove `str()` from these 2:
https://github.com/aio-libs/aiohttp/blob/master/tests/test_run_app.py#L521
https://github.com/aio-libs/aiohttp/blob/master/tests/test_run_app.py#L535

**Comment by SamirPS** [comment]
> Need to see a test covering this. Could probably just remove `str()` from these 2: https://github.com/aio-libs/aiohttp/blob/master/tests/test_run_app.py#L521 https://github.com/aio-libs/aiohttp/blob/master/tests/test_run_app.py#L535

Okay i remove str() for this two test and all test in this file passed

File: aiohttp/web.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import asyncio
 import logging
+import pathlib
```
Comment (Dreamsorcerer) [inline aiohttp/web.py]: Minor nitpick. In most places we do `from pathlib import Path`.

**Comment by Dreamsorcerer** [comment]
Also, test failing. Looks like there's another place that needs a `Union[]` annotation.

**Comment by SamirPS** [comment]
@Dreamsorcerer For me all test passed or skipped because i working on ubuntu (WSL)

File: aiohttp/client_exceptions.py [hunk 2]
```diff
@@ -175,13 +176,16 @@ class UnixClientConnectorError(ClientConnectorError):
...
+        path: Optional[Union[str, Path]],
+        connection_key: ConnectionKey,
+        os_error: OSError,
     ) -> None:
         self._path = path
         super().__init__(connection_key, os_error)
 
     @property
-    def path(self) -> str:
+    def path(self) -> Optional[Union[str, Path]]:
```
Comment (Dreamsorcerer) [inline aiohttp/client_exceptions.py:188]: I don't understand why you've just added `Optional` everywhere. You've now gone from 1 error to 32 errors.

**Comment by Dreamsorcerer** [comment]
We run a lot of tests in CI, so check the output. You should be able to reproduce the mypy errors on your machine, by simply running `mypy`.

File: aiohttp/client_exceptions.py (hunk 2 above)
Comment (Dreamsorcerer) [reply aiohttp/client_exceptions.py:188]: If you revert these changes, back to where you were before. There was only 1 error, it should be easy to then add the right annotation to pass the tests, you were almost there.

Comment (SamirPS) [reply aiohttp/client_exceptions.py:188]: i create a new pull request

**Pull request closed without merging**